test/upgrade/prober/forwarder.go

```diff
@@ -53,14 +59,14 @@ func (p *prober) deployForwarder(ctx context.Context) {
 	}
 }
 
-func (p *prober) removeForwarder() {
+func (p *Prober) removeForwarder() {
```

Comment:
same here, wire context through and don't create another one in L65. Not your doing, but since you're making changes to the interfaces already :)